src/v/coproc/tests/api/wasm_list_tests.cc

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "coproc/api/wasm_list.h"
+#include "coproc/tests/fixtures/coproc_slim_fixture.h"
+#include "coproc/tests/utils/coprocessor.h"
+#include "model/metadata.h"
+#include "reflection/adl.h"
+#include "test_utils/fixture.h"
+
+#include <boost/test/tools/old/interface.hpp>
+
+#include <algorithm>
+#include <numeric>
+#include <type_traits>
+
+using copro_typeid = coproc::registry::type_identifier;
+
+template<typename T>
+class rapidjson_value_comparator {
+public:
+    explicit rapidjson_value_comparator(T&& value)
+      : _value(std::forward<T>(value)) {}
+
+    bool operator()(const rapidjson::Value& v) const {
+        if constexpr (std::is_integral_v<T>) {
+            return v.GetInt() == _value;
+        }
+        if constexpr (
+          std::is_same_v<T, std::string> || std::is_same_v<T, ss::sstring>) {
+            return v.GetString() == _value;
+        }
+        if constexpr (std::is_same_v<T, bool>) {
+            return v.GetBool() == _value;
+        }
+        if constexpr (std::is_floating_point_v<T>) {
+            return v.GetDouble() == _value;
+        }
+        __builtin_unreachable();
+        return false;
+    }
```

Comment:
change this to statically assert that all type cases are covered. see reflection, serde, acl for some examples